cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ScheduleSpecificationCodec.java

```diff
@@ -47,7 +86,28 @@ public ScheduleSpecification deserialize(JsonElement json, Type typeOfT, JsonDes
     throws JsonParseException {
     JsonObject jsonObj = json.getAsJsonObject();
 
-    Schedule schedule = context.deserialize(jsonObj.get("schedule"), Schedule.class);
+    JsonElement scheduleTypeJson = jsonObj.get("scheduleType");
+    ScheduleType scheduleType;
+    if (scheduleTypeJson == null) {
+      // For backwards compatibility with spec persisted with older versions than 2.8, we need these lines
+      scheduleType = ScheduleType.ORIGINAL_TIME;
```

Comment:
I was thinking of SPECIFICATION_VERSION to be part of individual specifications, so that different specifications can have their own versions independently. e.g. `ScheduleSpecification` can have its own version say `1.0` and it can be used in `ScheduleSpecificationCodec` to serialize/deserialized to different fields.